pymove/utils/networkx.py

```diff
@@ -0,0 +1,338 @@
+    max_sampling_target: Number, optional, default 10
+        Maximum number of paths to be returned, considering the observed destination.
+    label_local: String, optional, default 'local_label'
+        Name of the column referring to the trajectories.
+    label_tid: String, optional, default 'tid_stat'
+        Column name for trajectory IDs.
+    simple_paths: Boolean, optional, default False
+        If true, use the paths with the most used sections.
+        Otherwise, use paths with less used sections.
+
+    """
+    source = str(source)
+    target = str(target)
+
+    if not nx.has_path(graph, source, target):
+        return []
+
+    if simple_paths:
+        all_paths = nx.all_simple_paths
+        param = max_path_size - 1
+
+    else:
+        all_paths = nx.shortest_simple_paths
+        param = None
+
+    for path in all_paths(graph, source, target, param):
+        freq_source = nx.get_node_attributes(graph, 'freq_source')[source]
+        freq_target = nx.get_node_attributes(graph, 'freq_target')[target]
+
+        if freq_source >= max_sampling_source:
+            break
+
+        if freq_target >= max_sampling_target:
+            break
+
+        if len(path) > max_path_size and simple_paths is False:
+            break
+
+        if len(path) >= min_path_size:
+            if path not in data[label_local].values.tolist():
+
+                append_trajectory(data, path, graph, label_tid)
+
+                freq_source += 1
+                freq_target += 1
+
+                graph.add_node(source, freq_source=freq_source)
+                graph.add_node(target, freq_target=freq_target)
+
+
+def graph_to_dict(graph: DiGraph) -> Dict:
+    """
+    Graph to Dict.
+
+    Converts nodes and edges from the Transition Graph
+    with all your attributes in a dictionary.
+
+    Parameters
+    ----------
+    graph: networkx.classes.digraph.DiGraph
+        Transition graph constructed from trajectory data.
+
+    Return
+    ------
+    Dict
+        Dictionary with the attributes of nodes and edges.
+    """
+    dict_graph = {'nodes': {}, 'edges': {}}
+
+    dict_graph['nodes']['coords'] = nx.get_node_attributes(graph, 'coords')
+    dict_graph['nodes']['datetime'] = nx.get_node_attributes(graph, 'datetime')
+    dict_graph['nodes']['freq_source'] = nx.get_node_attributes(graph, 'freq_source')
+    dict_graph['nodes']['freq_target'] = nx.get_node_attributes(graph, 'freq_target')
+    dict_graph['edges'] = nx.to_dict_of_dicts(graph)
+
+    return dict_graph
+
+
+def save_graph_as_json(
+    graph: DiGraph, filename: Optional[Text] = 'graph.json'
+):
+    """
+    Save Graph as JSON.
+
+    Saves the data extracted from the Transition Graph
+    into a JSON file.
+
+    Parameters
+    ----------
+    graph: networkx.classes.digraph.DiGraph
+        Transition graph constructed from trajectory data.
+    filename: String, Optional, default 'graph.json'
+        File name that will be saved with transition graph data.
```

Comment:
Modify to accept input of type `Path`. If type `str` convert to `Path`
Write as:
```
filename: str or Path, optional
    File name that will be saved with transition graph data, by default 'graph.json'
```